### ISSUE TITLE
Column `time_series` doesn't exist

### ISSUE THREAD

**Issue by agronomofiorentini** [e-sensing/sits#886]
**Describe the bug**
Dear creators,
When I proceed to use the `sits_get_data` the console gives me the following error output 

```
> points_training <- sits_get_data(cube = sinop_cube,
+                                  samples = training)
Error in `tidyr::unnest()`:
! Can't subset columns that don't exist.
✖ Column `time_series` doesn't exist.
Run `rlang::last_error()` to see where the error occurred.
```

so I tried to go deeper to this problem and I bring back the following outputs.

```
> rlang::last_error()
<error/vctrs_error_subscript_oob>
Error in `tidyr::unnest()`:
! Can't subset columns that don't exist.
✖ Column `time_series` doesn't exist.
---
Backtrace:
 1. sits::sits_get_data(cube = sinop_cube, samples = training)
 8. tidyr:::unnest.data.frame(., "time_series")
Run `rlang::last_trace()` to see the full context.
```

And more 

```
> rlang::last_trace()
<error/vctrs_error_subscript_oob>
Error in `tidyr::unnest()`:
! Can't subset columns that don't exist.
✖ Column `time_series` doesn't exist.
---
Backtrace:
     ▆
  1. ├─sits::sits_get_data(cube = sinop_cube, samples = training)
  2. ├─sits:::sits_get_data.data.frame(cube = sinop_cube, samples = training)
  3. │ ├─sits:::.sits_get_ts(...)
  4. │ └─sits:::.sits_get_ts.raster_cube(...)
  5. │   └─... %>% ...
  6. ├─dplyr::group_by(...)
  7. ├─tidyr::unnest(., "time_series")
  8. ├─tidyr:::unnest.data.frame(., "time_series")
  9. │ └─tidyselect::eval_select(enquo(cols), data)
 10. │   └─tidyselect:::eval_select_impl(...)
 11. │     ├─tidyselect:::with_subscript_errors(...)
 12. │     │ └─rlang::try_fetch(...)
 13. │     │   └─base::withCallingHandlers(...)
 14. │     └─tidyselect:::vars_select_eval(...)
 15. │       └─tidyselect:::as_indices_sel_impl(...)
 16. │         └─tidyselect:::as_indices_impl(...)
 17. │           └─tidyselect:::chr_as_locations(x, vars, call = call, arg = arg)
 18. │             └─vctrs::vec_as_location(...)
 19. └─vctrs (local) `<fn>`()
 20.   └─vctrs:::stop_subscript_oob(...)
 21.     └─vctrs:::stop_subscript(...)
 22.       └─rlang::abort(...)
```

**To Reproduce**
I was thinking the best to allow you to get the data to reproduce my problem, and I realized that the best way to be able to share, even though it's not the best is a shared drive folder.

https://drive.google.com/file/d/118VrhYqE26a8AhqfWi8Y4dJK2dNBIM3W/view?usp=share_link

<details>
Paste the output of your `R version 4.2.0 (2022-04-22 ucrt)
Platform: x86_64-w64-mingw32/x64 (64-bit)
Running under: Windows 10 x64 (build 22000)

Matrix products: default

locale:
[1] LC_COLLATE=Italian_Italy.utf8  LC_CTYPE=Italian_Italy.utf8    LC_MONETARY=Italian_Italy.utf8 LC_NUMERIC=C                   LC_TIME=Italian_Italy.utf8    

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] sits_1.2.0

loaded via a namespace (and not attached):
 [1] pkgload_1.3.2        tidyr_1.2.1          bit64_4.0.5          warp_0.2.0           shiny_1.7.4          assertthat_0.2.1     sp_1.5-0            
 [8] yaml_2.3.6           remotes_2.4.2        sessioninfo_1.2.2    lattice_0.20-45      pillar_1.8.1         glue_1.6.2           torch_0.9.0         
[15] digest_0.6.31        promises_1.2.0.1     htmltools_0.5.4.9000 httpuv_1.6.7         pkgconfig_2.0.3      devtools_2.4.5       raster_3.6-11       
[22] purrr_0.3.5          xtable_1.8-4         processx_3.8.0       slider_0.3.0         terra_1.6-47         later_1.3.0          timechange_0.1.1    
[29] tibble_3.1.8         proxy_0.4-27         generics_0.1.3       usethis_2.1.6        ellipsis_0.3.2       cachem_1.0.6         withr_2.5.0         
[36] cli_3.5.0            magrittr_2.0.3       crayon_1.5.2         mime_0.12            memoise_2.0.1        ps_1.7.2             fs_1.5.2            
[43] fansi_1.0.3          class_7.3-20         pkgbuild_1.4.0       profvis_0.3.7        tools_4.2.0          data.table_1.14.6    prettyunits_1.1.1   
[50] lifecycle_1.0.3      stringr_1.5.0        callr_3.7.3          compiler_4.2.0       e1071_1.7-12         rlang_1.0.6          classInt_0.4-8      
[57] units_0.8-1          grid_4.2.0           rstudioapi_0.14      htmlwidgets_1.6.0    crosstalk_1.2.0      miniUI_0.1.1.1       leafem_0.2.0        
[64] base64enc_0.1-3      codetools_0.2-18     DBI_1.1.3            curl_4.3.3           R6_2.5.1             lubridate_1.9.0      dplyr_1.0.10        
[71] fastmap_1.1.0        bit_4.0.5            utf8_1.2.2           rprojroot_2.0.3      coro_1.0.3           KernSmooth_2.23-20   desc_1.4.2          
[78] stringi_1.7.8        parallel_4.2.0       Rcpp_1.0.9           vctrs_0.5.1          sf_1.0-9             png_0.1-8            leaflet_2.1.1       
[85] tidyselect_1.2.0     urlchecker_1.0.1    `
</details>

If you need any other information, i am at your disposal

**Comment by rolfsimoes** [comment]
I was able to reproduce your error. 
The problem occurs due to the projection of the coordinates in the CSV. By default, sits assumes that the coordinates are in WGS-84. 
We can use the `crs` parameter of the `sits_get_data()` function to enter a different CRS. We took that the coordinates of the points were the same as the cube projection and the example below worked.
```R
library(sits)
sinop_cube <- sits_cube(
    source = "BDC", 
    collection = "MOD13Q1-6", 
    parse_info = c("tile", "band", "date"),
    data_dir = "~/../Downloads/github_issue_sits/github_issue_sits/ndvi/"
)

training <- "~/../Downloads/github_issue_sits/github_issue_sits/csv/training.csv"
points_training <- sits_get_data(
    cube = sinop_cube,
    samples = training,
    crs = unique(sinop_cube$crs)
)

```

We will improve the checks of the `sits_get_data()` function to provide a more user-friendly and useful error message. Thank you for bringing this problem to our attention.

**Comment by agronomofiorentini** [comment]
It is working perfectly.

Thanks

**Comment by agronomofiorentini** [comment]
I'm sorry to bother again.
But I tried to do the whole procedure of training and classifying the image but I get a really strange output.
I attach the classification that happened

<img width="587" alt="Screenshot_1" src="https://user-images.githubusercontent.com/66969646/209554053-dcbbc62a-9d8c-4b31-9fd5-30693802afc4.png">

I have controlled the dataset and this is the driver folder that allow you to have the same dataset that i am using
https://drive.google.com/drive/folders/1-5H535UOJ4GuvR9Zl189-i82D7knH5L_?usp=share_link 

All the procedure that i have done is the following

```
library(sits)
sinop_cube <- sits_cube(
    source = "BDC", 
    collection = "MOD13Q1-6", 
    parse_info = c("tile", "band", "date"),
    data_dir = "~/../Downloads/github_issue_sits/github_issue_sits/ndvi/"
)

training <- "~/../Downloads/github_issue_sits/github_issue_sits/csv/training.csv"
points_training <- sits_get_data(
    cube = sinop_cube,
    samples = training,
    crs = unique(sinop_cube$crs)
)

tcnn_model <- sits_train(samples = points_training,
                         ml_method = sits_tempcnn(verbose = TRUE))

predicted <- sits_classify(sinop_cube, tcnn_model)

raster_probs_bayes <- sits_smooth(predicted,
                                  type = "bayes")

label_cube <- sits_label_classification(raster_probs_bayes)

plot(label_cube)
```

I don't understand where the problem is also because I think I performed everything correctly.

In addition, to increase the possibility of being able to reproduce the entire problem, I also used the docker image 
 https://hub.docker.com/r/brazildatacube/sits-rstudio

if you want we can also close this issue and open another one, of course if you think it is more correct

**Comment by rolfsimoes** [comment]
I was able to reproduce your classification. 

This issue occurs due to a bug in the previous version of sits (1.1.0) for images that have no scaling factor.

In the new version (1.2.0), we have fixed this bug. If you upgrade to the latest sits version, this problem should be solved. See the classified map that I have produced:

![image](https://user-images.githubusercontent.com/9295371/210667193-2af9b276-a78a-4caa-b551-482562361c23.png)
